addons/survey/static/src/js/survey_form.js

```diff
@@ -0,0 +1,370 @@
+odoo.define('survey.form', function(require) {
+'use strict';
+
+var ajax = require('web.ajax');
+var field_utils = require('web.field_utils');
+var rpc = require('web.rpc');
+var publicWidget = require('web.public.widget');
+var time = require('web.time');
+
+publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
+    selector: '.o_survey_form',
```

Comment:
One thing I would do: explicitely have a readonly flagon the survey widget, coming from data (data-readonly), set in template (taking survey, readonly=False, results page, readonly=True).

That way in the code we can do if this.readonly ... to better understand there is some code specific for readonly (or for not-readonly).